src/views/ChargeLog.vue

```diff
@@ -489,6 +490,11 @@ export default {
       const baseUrl = `${location.protocol}//${location.hostname}:${port}/openWB/web/settings/downloadChargeLog.php`;
       return baseUrl + `?year=${this.chargeLogRequestData.year}&month=${this.chargeLogRequestData.month}`;
     },
+    downloadUrlYear() {
+      const port = parseInt(location.port) || (location.protocol == "https:" ? 443 : 80);
+      const baseUrl = `${location.protocol}//${location.hostname}:${port}/openWB/web/settings/downloadChargeLog.php`;
```

Comment:
As we have two URLs now, the property `downoadURL()` should be renamed to something more meaningful like `downloadUrlMonth()`.

Please do not duplicate code. Create a new computed property for the `baseURL` and make use of it in `downloadUrlMonth()` and `downloadUrlYear()`.

Reply:
@Sleepwalker86 just a reminder